src/runtime/providers/math.ts

```diff
@@ -13,7 +13,9 @@ abstract class TotalFunctionConstraint extends Constraint {
   // proposed variable.
   resolveProposal(proposal, prefix) {
     let {args} = this.resolve(prefix);
-    return [this.getReturnValue(args)];
+    let result = this.getReturnValue(args);
+    if (isNaN(result) || !(isFinite(result))) {return [];}
```

Comment:
Inifinity is a well-defined value, so I think we can remove that check and just have the NaN one.